test/pylib/driver_util.py

```diff
@@ -0,0 +1,23 @@
+"""Helper utilities to work around Scylla/Cassandra driver issues,
+   specifically not reconnecting automatically after a host or a set of
+   hosts were restarted."""
+# Copyright (C) 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import logging
+from test.pylib.manager_client import ManagerClient
+from cassandra.cluster import Session                   # type: ignore # pylint: disable=no-name-in-module
+
+
+async def reconnect_driver(manager: ManagerClient) -> Session:
```

Comment:
BTW. `scylladb/python-driver#170` is already fixed and included in the dbuild docker image.
Driver should always reconnect if IPs don't change, and using `wait_for_cql` and `wait_for_cql_and_get_hosts` should suffice to ensure that the driver reconnects to a given host.

I just didn't get to removing this `reconnect_driver` workaround from Raft upgrade tests.

But the driver won't handle changing the IPs of all nodes. Then it may simply be impossible for the driver to reconnect - it's trying to use old IPs after all, in which case we need to connect it manually using a new set of IPs. But that's not a bug, it's expected.

In that case we should probably update the docstring, explaining when we may need to reconnect the driver.

Reply:
Also maybe it's better to keep those utils in one place? We have test/pylib/utils.py and some of the functions there are already related to the driver.

Reply:
2. I dropped the patch from this series. I will take a look if it is still necessary for test_change_all

Reply:
> I will take a look if it is still necessary for test_change_all

It probably is, the driver doesn't know the new IPs of the cluster. If at least one node does not change its IP the driver should fetch IPs of others from there, but not if every node changes their IPs.

My point is, in this scenario this is no longer a workaround for scylladb/python-driver#170.